### PR TITLE
TDE filter implemented with other filters

### PR DESCRIPTION
**Pull request by davidlmoutard**
Added TDE filter to same part of code as other filters, along with some necessary (but slightly clunky) methods to convert input format. Added logic to print which filters were passing spectra. Minor fixes to TDE filter itself, notably including a line to silence a common runtime warning which I am already managing.

### CODE REVIEW

File: desidiff/nb/desidiff.ipynb [hunk 1]
```diff
@@ -168,17 +172,38 @@
     "        # Difference spectrum may have high-frequency signal\n",
     "        perres_filter = perconv_SN(difflux_clipmean,difivar,difmask)\n",
     "        \n",
+    "        #Turns diffwave and difflux into 1 list each instead of by filter for TDE check.\n",
```
Comment (AlexGKim) [inline desidiff/nb/desidiff.ipynb:175]: This operation of combining the spectra should be inside of TDE Check

**Review by AlexGKim — COMMENTED** [review]
To keep the top-level code clean, put this combination somewhere else.

What I recommend is writing a separate function that does the combination.  Then have the TDE call that function.

**Review by AlexGKim — COMMENTED** [review]
Notebooks should have cells cleared for pull requests.

**Comment by davidlmoutard** [comment]
> Notebooks should have cells cleared for pull requests.

Sorry about that- I must have clicked "Clear Output" instead of "Clear all outputs"

File: desidiff/nb/desidiff.ipynb [hunk 2]
```diff
@@ -146,7 +147,9 @@
     "        difmask = {key: newmask[key] + refmask[key]\n",
     "                       for key in newmask.keys()}\n",
     "        difwave = dict(newwave)\n",
-    "\n",
+    "        \n",
+    "        redshift = z_data[0][1] #it seems each set only has one value\n",
```
Comment (AlexGKim) [inline desidiff/nb/desidiff.ipynb:151]: What is this doing?

**Review by AlexGKim — APPROVED** [review]
Looks good but there is a comment on line 150

File: desidiff/nb/desidiff.ipynb (hunk 2 above)
Comment (davidlmoutard) [reply desidiff/nb/desidiff.ipynb:151]: This gets the redshift data which I pulled from the z_best files. It's a little dumb right now, but every iteration only has 1 targetID, and so z_data is a length 1 array, containing a tuple with TARGETID, Z, ZERR, ZWARN, and SPECTYPE. z_data[0][1] is just the redshift for that targetID, which I need to shift the spectra into the rest frame

**Pull request merged**